### PR TITLE
Enhancement: Fixed models.py and piechart.py

### LINKED ISSUE #109: Enhancement: Password feature implemented and support for all file extensions.
@this-is-yaash 
- While working on the project I found out that we can add password protection in `piechart.py` too. Here is my approach to dynamically set a password which will keep changing with the time.
```python
def plot(snippet_name,password):
    current_time = datetime.now().strftime("%H%M")
    if str(password) != current_time:
        raise ValueError("syntax error: incorrect password")
```

- I have also planned to add support for all the extensions using, glob library which is inbuilt  and present in python standard library. Here is my approach for that.

```python
    try:
        base_dir = os.path.dirname(__file__)
        snippets_dir = os.path.join(base_dir, "stash")
        pattern = os.path.join(snippets_dir, f"{snippet_name}.*")

        matching_files = glob.glob(pattern)
        
```
- I can also add a check to see if there is no file or multiple files with the same name in the directory. Here is my approach to this.

```python
        if not matching_files:
            raise FileNotFoundError(f"No file found with the name.")
        elif len(matching_files) > 1:
            raise ValueError("Multiple files found with the given name.")
```

- I can do error handling in it by putting the main features in a try block and I can also handle all possible errors in the code. here is my approach to this.

```python

def func(args):
     try:
          #code 
     except:
          #code 
  ```
  
  please assign me this issue under gssocext.
Thank you.

### PR DESCRIPTION
**Pull request by tannuiscoding**
### List of things changed.
- [x] Have you renamed the PR Title in a meaningful way?

### Related Issue

closes #109 

Enhanced the working of `model.py` and `piechart.py`.
 
### Description

- `models.py` now has the feature of copying to clipboard and displaying source on the console.
- `piechart.py` now has password protection
- `model.py` now has password protection.

### Type of change

What sort of changes have you made:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### Checklist

- [x] My code follows the code style of this project.
- [x] I have followed the contribution guidelines
- [x] I have performed a self-review of my own code.
- [x] I have ensured my changes don't generate any new warnings or errors.
- [x] I have updated the documentation (if necessary).
- [x] I have resolved all merge conflicts.

### CODE REVIEW

**Comment by ShashwatAgrawal20** [comment]
hey @tannuiscoding, a test case is failing can you fix that too.

File: package/graph.py [hunk 1]
```diff
@@ -5,17 +5,19 @@
 import shutil
 
 def display(snippet_name, password):
-    current_time =  datetime.now().strftime("%H%M")
+    # Retrieve the current time in HHMM format
+    current_time = datetime.now().strftime("%H%M")
+    # Check if the provided password matches the current time
     if str(password) != current_time:
         raise ValueError("syntax error")
+    # Proceed to copy code to clipboard if the password matches
```
Comment (ShashwatAgrawal20) [inline package/graph.py:13]: ig we don't need these comments.

File: package/test.py
Comment (ShashwatAgrawal20) [inline package/test.py:1]: Do we need this file, I don't think so?

**Review by ShashwatAgrawal20 — CHANGES_REQUESTED** [review]
Might have missed some things, and I see a test cases failing can you fix that too.

File: package/test.py
Comment (this-is-yaash) [reply package/test.py:1]: Yes, delete them and make PR

**Pull request merged**